statement_test.go

```diff
@@ -630,3 +631,67 @@ func retrieveRows(rows *sql.Rows, ch chan string) {
 	ch <- s
 	close(ch)
 }
+
+func TestEmitQueryID(t *testing.T) {
+	var db *sql.DB
+	var err error
+
+	if db, err = sql.Open("snowflake", dsn); err != nil {
+		t.Fatalf("failed to open db. %v, err: %v", dsn, err)
+	}
+	defer db.Close()
+
+	queryIDChan := make(chan string, 1)
+	numrows := 100000
+	ctx, _ := WithAsyncMode(context.Background())
+	ctx = WithQueryIDChan(ctx, queryIDChan)
+	conn, _ := db.Conn(ctx)
+
+	dest := make([]driver.Value, 2)
+
+	goRoutineChan := make(chan string)
+	go func(grCh chan string, qIDch chan string) {
+		queryID := <- queryIDChan
+		grCh <- queryID
+	}(goRoutineChan, queryIDChan)
+
+	err = conn.Raw(func(x interface{}) error {
+		stmt, err := x.(driver.ConnPrepareContext).PrepareContext(ctx, fmt.Sprintf("SELECT SEQ8(), RANDSTR(1000, RANDOM()) FROM TABLE(GENERATOR(ROWCOUNT=>%v))", numrows))
+		if err != nil {
+			return err
+		}
+		rows, err := stmt.(driver.StmtQueryContext).QueryContext(ctx, nil)
+		if err != nil {
+			return err
+		}
+		defer rows.Close()
+
+		if err != nil {
+			return err
+		}
+
+		cnt := 0
+		for {
+			err = rows.Next(dest)
+			if err != nil {
+				if err == io.EOF {
+					break
+				} else {
+					t.Fatalf("unexpected error: %v", err)
+				}
+			}
+			cnt++
+		}
+		queryID := <- goRoutineChan
```

Comment:
If no queryId is returned, what is the error the caller has to handle? A test case on such a error situation will be helpful.